irodsConnector/session.py

```diff
@@ -171,14 +171,6 @@ def connect(self):
 
         """
         logging.debug(f'{self.irods_env_file=}')
-        if not self.irods_env_file:
```

Comment:
This works fine.  Is it simply undesired functionality?

Reply:
I know and I like the feature, I just think that the frontend (GUI or Cli) which check all parameters before giving them to the backend should decide which irods_environment to use, the previous one or if the user says so, the new one. The backend should not "correct" the users' choices, that should be up to the frontend.

Reply:
Perhaps it is an idea to add an option to the configuration to allow a user to choose this behavior?  Something like a boolean value to `"use_prev_ienv": False`.

I like the idea of `"last_ienv": ""` to indicate this, but the setting of that is automatic and I don't see that changing.  Also, given that "last_ienv" gives the default environment file selection in the GUI, a new mechanism for that would be needed.